torch/csrc/autograd/function.h

```diff
@@ -122,7 +108,7 @@ struct Function : std::enable_shared_from_this<Function> {
 
   bool should_compute_output(size_t index) const {
     TORCH_ASSERTM(index < next_functions.size(), "Index out of range");
-    return next_functions[index].first != nullptr;
+    return next_functions[index].function != nullptr;
```

Comment:
It looks like `operator bool()` on `Edge`s might be useful (you don't have to implement it, just a general observation)

Reply:
Makes sense. I see 3-4 places where we check if `function` is `nullptr`. I'll give `Edge` an `is_valid()` method, because I feel like `if (edge) { ... }` is not as immediately obvious as `if (edge.is_valid()) { ... }`.